src/core/transient_variable.jl

```diff
@@ -0,0 +1,295 @@
+    nw::Int = gm.cnw;
+    bounded::Bool = true,
+    report::Bool = true,
+)
+    rho =
+        var(gm, nw)[:density] = JuMP.@variable(
+            gm.model,
+            [i in ids(gm, nw, :junction)],
+            base_name = "$(nw)_rho",
+            start = comp_start_value(
+                ref(gm, nw, :junction),
+                i,
+                "rho_start",
+                ref(gm, nw, :junction, i)["p_min"],
+            )
+        )
+
+    if bounded
+        for (i, junction) in ref(gm, nw, :junction)
+            JuMP.set_lower_bound(rho[i], junction["p_min"])
+            JuMP.set_upper_bound(rho[i], junction["p_max"])
+        end
+    end
+
+    report &&
+    _IM.sol_component_value(gm, nw, :junction, :density, ids(gm, nw, :junction), rho)
+    report &&
+    _IM.sol_component_value(gm, nw, :junction, :pressure, ids(gm, nw, :junction), rho)
+end
+
+"variables associated with compressor mass flow (transient)"
+function variable_compressor_flow(
+    gm::AbstractGasModel,
+    nw::Int = gm.cnw;
+    bounded::Bool = true,
+    report::Bool = true,
+)
+    max_mass_flow = ref(gm, nw, :max_mass_flow)
+    f =
+        var(gm, nw)[:compressor_flow] = JuMP.@variable(
+            gm.model,
+            [i in ids(gm, nw, :compressor)],
+            base_name = "$(nw)_f_compressor",
+            start = comp_start_value(
+                ref(gm, nw, :compressor),
+                i,
+                "f_compressor_start",
+                0.0,
+            )
+        )
+
+    if bounded
+        for (i, compressor) in ref(gm, nw, :compressor)
+            JuMP.set_lower_bound(f[i], -max_mass_flow)
+            JuMP.set_upper_bound(f[i], max_mass_flow)
+        end
+    end
+
+    report &&
+    _IM.sol_component_value(gm, nw, :compressor, :flow, ids(gm, nw, :compressor), f)
+end
+
+"variables associated with pipe flux (transient)"
+function variable_pipe_flux(
+    gm::AbstractGasModel,
+    nw::Int = gm.cnw;
+    bounded::Bool = true,
+    report::Bool = true,
+)
+    phi =
+        var(gm, nw)[:pipe_flux] = JuMP.@variable(
+            gm.model,
+            [i in ids(gm, nw, :pipe)],
+            base_name = "$(nw)_flux_pipe",
+            start = comp_start_value(
+                ref(gm, nw, :pipe),
+                i,
+                "phi_start",
+                ref(gm, nw, :pipe, i)["flux_min"],
+            )
+        )
+
+    if bounded
+        for (i, pipe) in ref(gm, nw, :pipe)
+            JuMP.set_lower_bound(phi[i], pipe["flux_min"])
+            JuMP.set_upper_bound(phi[i], pipe["flux_max"])
+        end
+    end
+
+    if report
+        _IM.sol_component_value(gm, nw, :pipe, :flux, ids(gm, nw, :pipe), phi)
+        sol_f =
+            Dict(i => phi[i] * ref(gm, nw, :pipe, i)["area"] for i in ids(gm, nw, :pipe))
+        _IM.sol_component_value(gm, nw, :pipe, :flow, ids(gm, nw, :pipe), sol_f)
+    end
+end
+
+"variables associated with compression ratio (transient)"
+function variable_c_ratio(
```

Comment:
In general, I'd like to see the transient model reusing as much as possible from the steady-state model, so if this can directly use the compression ratio variable in ``variable.jl`` lets do that.  In the long run, esp. as we get a steady-state more in the density/flux space we'll have alot of overlap and I'd like to leverage that is much as possible to keep the code small and agile.

Reply:
Agree, but as @pseudocubic said, there is no compressor ratio variable in variable.jl. There is only compressor ratio squared.

Reply:
Refactored the function name in variable.jl to ``variable_compressor_ratio_sqr`` and ``variable_compresor_ratio_sqr_value``